File: data_structures/queue/linearqueue_using_array.js
const input = require("prompt-sync")();
 
class queue{
    constructor(size=5){
        this.size=size;
        this.front=0;
        this.rear=-1;
        this.array= new Array(size);
    }
    push(value){
        if(this.rear==this.size-1){
            console.log("queue is full");
            return 0;
        }
        else{
            this.rear++;
            this.array[this.rear]=value;
            console.log(value, " is added in queue");
            console.log("front: ",this.front);
            console.log("rear: ",this.rear);
            return 1;
        }
    }
    pop(){
        if(this.front>this.rear){
            console.log("queue is empty");
            return 0;
        }
        else{
            console.log(this.array[this.front], " is poped");
            this.front++;
            console.log("front: ",this.front);
            console.log("rear: ",this.rear);
            return 1;

        }
    }

    display(){
        if(this.front>this.rear){
            console.log("queue is empty");
            return 0;
        }
        else{
            for(var i=this.front;i<=this.rear;i++){
                console.log(this.array[i]);
            }
        }
    }
}


var qu =new queue();
console.log("1: push \n2: pop \n3: display\n4: exit");
while(true){
    var x=input("enter choice: ");
    if(x==1){
        var value = Number(input("enter value: "));
        qu.push(value);
    }
    else if(x==2){
        qu.pop();
    }
    else if(x==3){
        qu.display();
    }
    else{
        break;
    }
}

File: data_structures/queue/linearqueue_using_array.c
#include <stdio.h>
#include <stdlib.h>

struct queue
{int size,front,rear,*Q;};

void creatqueue(struct queue *qu){
	printf("\nenter the size of queue: ");
	scanf("%d",&qu->size);
	qu->front=0;
	qu->rear=-1;
	qu->Q=(int *)malloc(qu->size*sizeof(int));
}

void push(struct queue *qu,int value){
	if(qu->rear==qu->size-1){
		printf("\nqueue is full");
	}
	else{
		qu->rear++;
		qu->Q[qu->rear]=value;
		printf("%d id added in queue\n",value);
		printf("\nfront : %d  rear: %d\n",qu->front,qu->rear);
	}
}

int pop(struct queue *qu){
	int x=0;
	if(qu->front>qu->rear){
		printf("\nqueue is empty");
	}
	else{
		x=qu->Q[qu->front];
		printf("\n%d is poped\n", qu->Q[qu->front]);
		qu->front++;
		
	}
	return x;

}

void display(struct queue *qu){
	for(int i=qu->front;i<=qu->rear;i++){
		printf("%d\n", qu->Q[i]);
	}
}

int main()
{
	struct queue qu;
	creatqueue(&qu);
	while(1){
		printf("\npress 1 for push\npress 2 for pop\npress 3 for display\npress 4 for exit\nenter choice: ");
		int i;
		scanf("%d",&i);
		if (i==1)
		{
			int value;
			printf("enter the value: ");
			scanf("%d",&value);
			push(&qu,value);
		}
		else if (i==2)
		{
			pop(&qu);
		}
		else if (i==3)
		{
			display(&qu);
		}
		else
		{
			break;
		}
	}
	return 0;
}
